resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesUtils.scala

```diff
@@ -216,6 +217,32 @@ private[spark] object KubernetesUtils extends Logging {
     Hex.encodeHexString(random) + time
   }
 
+  /**
+   * This function builds the Quantity objects for each resource in the Spark resource
+   * configs based on the component name(spark.driver.resource or spark.executor.resource).
+   * It assumes we can use the Kubernetes device plugin format: vendor-domain/resource.
+   * It returns a set with a tuple of vendor-domain/resource and Quantity for each resource.
+   */
+  def buildResourcesQuantities(
+      componentName: String,
+      sparkConf: SparkConf): Set[(String, Quantity)] = {
+    val allResources = sparkConf.getAllWithPrefix(componentName)
```

Comment:
is this corresponding to `spark.{driver,executor}.resources...`

Reply:
yes

Reply:
what kind of strings are in `componentName`? It seems like this call only works if the strings start with `spark.driver.resources.<name>...`

Reply:
As the javadoc on the function says "component name(spark.driver.resource or spark.executor.resource)."
this fucntion is called from both the driver and the executor feature steps. one passes SPARK_DRIVER_RESOURCE_PREFIX and one passes SPARK_EXECUTOR_RESOURCE_PREFIX

Also note we have another jira to abstract some of the resource configs about to make it more readable - SPARK-27823